persistence/google-cloud-spanner/src/main/resources/spanner/schema-v1.sql

```diff
@@ -0,0 +1,86 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one
+-- or more contributor license agreements.  See the NOTICE file--
+--  distributed with this work for additional information
+-- regarding copyright ownership.  The ASF licenses this file
+-- to you under the Apache License, Version 2.0 (the
+-- "License"). You may not use this file except in compliance
+-- with the License.  You may obtain a copy of the License at
+--
+--  http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing,
+-- software distributed under the License is distributed on an
+-- "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+-- KIND, either express or implied.  See the License for the
+-- specific language governing permissions and limitations
+-- under the License.
+
+CREATE SEQUENCE IF NOT EXISTS Ids OPTIONS(sequence_kind = 'bit_reversed_positive');
+
+CREATE TABLE IF NOT EXISTS Realms(
+  RealmId STRING(MAX) NOT NULL
+)
+PRIMARY KEY (RealmId);
+
+CREATE TABLE IF NOT EXISTS Entities(
+RealmId STRING(MAX) NOT NULL,
+CatalogId INT64 NOT NULL,
+Id INT64 NOT NULL,
+ParentId INT64 NOT NULL,
+Name STRING(MAX),
+EntityVersion INT64 NOT NULL,
+TypeCode INT64 NOT NULL,
+SubTypeCode INT64 NOT NULL,
+CreateTimestamp INT64,
+DropTimestamp INT64,
+PurgeTimestamp INT64,
+ToPurgeTimestamp INT64,
+LastUpdateTimestamp INT64,
+`Properties` JSON,
+InternalProperties JSON,
```

Comment:
any reason we want JSON over JSONB seems like spanner supports it : https://docs.cloud.google.com/spanner/docs/working-with-jsonb

Reply:
Spanner didn't support JSONB in Spanner mode (just PostgreSQL compatibility mode) when this was originally written. :-)